pkg/awsutils/awsutils.go

```diff
@@ -1515,3 +1520,25 @@ func (cache *EC2InstanceMetadataCache) IsUnmanagedENI(eniID string) bool {
 	}
 	return false
 }
+
+func (cache *EC2InstanceMetadataCache) getENIsFromPaginatedDescribeNetworkInterfaces(
+	input *ec2.DescribeNetworkInterfacesInput) ([]*ec2.NetworkInterface, error) {
+	outputENIs := make([]*ec2.NetworkInterface, 0)
+	pageNum := 0
+	log.Debugf("Paginating describe ENI has page size: %d", *input.MaxResults)
+	pageFn := func(output *ec2.DescribeNetworkInterfacesOutput, lastPage bool) (nextPage bool) {
+		pageNum++
+		log.Infof("EC2 DescribeNetworkInterfaces succeeded with %d results on page %d",
+			len(output.NetworkInterfaces), pageNum)
+
+		outputENIs = append(outputENIs, output.NetworkInterfaces...)
+		// if no more ENI, we want to exit even it is not the last page to avoid many blank calls.
+		if len(output.NetworkInterfaces) == 0 {
```

Comment:
You should always use token/continue fields in the output object. For https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#DescribeNetworkInterfacesOutput use 'NextToken'

Reply:
Makes sense. Have synced with EC2. The same suggestion. Thanks @SaranBalaji90.

Reply:
Updated to use nextToken != nill